commercesearch/infrastructure/productsearch/inmemoryrepository.go

```diff
@@ -107,7 +109,7 @@ func (r *InMemoryProductRepository) UpdateProducts(_ context.Context, products [
 		}
 		r.marketplaceCodeIndex[product.BaseData().MarketPlaceCode] = product
```

Comment:
maybe use the marketPlaceCode variable for indexing instead

Reply:
good point